experimental/runtimes/src/strategies/default.rs

```diff
@@ -14,9 +15,11 @@ pub struct DefaultThreadManager {
 
 impl DefaultThreadManager {
     pub(crate) fn new() -> DefaultThreadManager {
-        let exe_threads = spawn_rayon_thread_pool("exe".into(), Some(num_cpus::get()));
-        let non_exe_threads = spawn_rayon_thread_pool("non_exe".into(), Some(num_cpus::get()));
-        let io_threads = spawn_rayon_thread_pool("io".into(), Some(64));
+        // Do not use more than 32 threads for rayon thread pools as we have seen scalability issues with more threads.
+        // This needs to be revisited once we resolve the scalability issues.
+        let exe_threads = spawn_rayon_thread_pool("exe".into(), Some(min(num_cpus::get(), 32)));
+        let non_exe_threads = spawn_rayon_thread_pool("non_exe".into(), Some(min(num_cpus::get(), 32)));
+        let io_threads = spawn_rayon_thread_pool("io".into(), Some(32));
```

Comment:
We probably want to leave IO threads to a higher number, when it hits IO you will need more threads to get enough thoughput. also tasks in this pool tent to be larger, so the overhead ratio should be smaller.

Reply:
I doubt having higher number of these threads are helping even when the DB is large - because even in that case I assume we are CPU bound. Can you help run the large DB benchmark you have in mind and see if we are seeing any regression?

Reply:
with a very little amount of IO (account creation on 100M db on a 128G RAM machine), the put_stats_and_indices which uses this pool is ~15% to ~20% slower.. I'd expect in real world it would be worse.

Reply:
How about we run the large DB benchmark on this change - if that is significantly slower, we can bump this?

Reply:
If you only look at TPS you probably won't see the difference because the execution is slower today (and because the large DB benchmark is not really large, unless you run it on a smaller RAM machine), but conceptually this is not something I want to put in production. 

BTW I also did a side by side comparison for 32 v.s. 64 on small db for p2p, and I don't really see improvement. Maybe it just implicitly slow down the commit slightly and speed up execution slightly..